### PR TITLE
Add example on how to generate specs dinamically

### PR DESCRIPTION
**Pull request by fotanus**
As @sol suggested on #236, here is an example on the documentation

### CODE REVIEW

File: doc/writing-specs.markdown [hunk 1]
```diff
@@ -158,3 +158,20 @@ The report produced at the end of a test run counts passed, failed, and pending
...
+```hspec
+main :: IO ()
+main = hspec $ do
+  describe "generate specs" $ do
+    allFiles <- runIO (getDirectoryContents myDirPath)
+    let files = filter (`notElem` [".", ".."]) allFiles
+    forM_ files $ \file -> do
+      it ("test using the file " ++ file) $ do
+        pending
+```
```
Comment (sol) [inline doc/writing-specs.markdown]: Can you try to make this example executable, similar to what we already have under `doc/_includes/`?

**Comment by sol** [comment]
I think it makes sense to mention the implications of `runIO`, that it's run on spec construction time and also on e.g. `--dry-run`, and should not be used carelessly.  Some combination of what I said in #236 and what we have in the Haddocks.

**Comment by fotanus** [comment]
Sure, I'll do this later today, when I get home.

**Comment by fotanus** [comment]
like this?

**Comment by sol** [comment]
@fotanus sorry for not getting back to you. Are you still interested to get this on `master`? If yes, I'll leave another round of comments that need to be addressed.

File: doc/_includes/DynamicIts.hs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+main :: IO ()
+main = hspec $ do
+  describe "generate specs" $ do
+    allFiles <- runIO (getDirectoryContents ".")
+    let files = filter (`notElem` [".", ".."]) allFiles
+    forM_ files $ \file -> do
+      it ("test using the file " ++ file) $ do
+        pending
+
```
Comment (sol) [inline doc/_includes/DynamicIts.hs]: please remove that empty line

File: doc/writing-specs.markdown [hunk 3]
```diff
@@ -158,3 +158,27 @@ The report produced at the end of a test run counts passed, failed, and pending
 spec items separately, and having unresolved pending spec items does not cause
 your test suite to fail; that is, it can still exit with a status code of 0 if
 there are unresolved pending spec items.
+
+### Generating your specs dinamically
```
Comment (sol) [inline doc/writing-specs.markdown]: typo: dinamically -> dynamically

File: doc/writing-specs.markdown [hunk 4]
```diff
@@ -158,3 +158,27 @@ The report produced at the end of a test run counts passed, failed, and pending
...
+main = hspec $ do
+  describe "generate specs" $ do
+    allFiles <- runIO (getDirectoryContents ".")
+    let files = filter (`notElem` [".", ".."]) allFiles
+    forM_ files $ \file -> do
+      it ("test using the file " ++ file) $ do
+        pending
+```
+
+Note that tour runIO action is alway executed when the test tree is constructed,
```
Comment (sol) [inline doc/writing-specs.markdown]: tour -> your

File: doc/_includes/DynamicIts.hs (hunk 2 above)
Comment (sol) [inline doc/_includes/DynamicIts.hs]: This example is still not executed nor included into the documentation. To make that happen you would need to include it somewhere, in a very similar way to what is done for the other examples.

**Pull request closed without merging**